File: C/strindex.c
/*
$Id: strindex.c 5 2007-03-31 15:04:06Z Makhtar $
Get the index of a string
*/

int strindex(char s[], char t[]) {
  int i, j, k;

  /* move index from the beginning till
     the end of each line  */
  for (i = 0; s[i] != '\0'; i++) {
    /* move index from the beginning of the target  */
    for (j = i, k = 0; t[k] != '\0' && s[j] == t[k]; j++, k++)
      continue;

    if (k > 0 && t[k] == '\0')
      return i;
  }
  /* complain if we get here */
  return (-1);
}

File: Go/strindex.go
package main

import ()

func strindex(s string, t string) int {
	var i, j, k int
	var end byte
	println("Checking string", t, "in", s, end)

	for i = 0; i < len(s); i++ {
		j = i
		k = 0
		for j < len(s) && (s[j] != t[k]) {
			println(i, j, s[i])
			j++
			// continue;
		}
		k++

		if s[j-1] == t[k-1] {
			return i
		}
	}

	return -1
}

func main() {
	println(strindex("Hello", "o"))

}
